Sources/IO/XML/XMLWriter/index.js

```diff
@@ -0,0 +1,138 @@
+// Global methods
+// ----------------------------------------------------------------------------
+
+function compressBlock(uncompressed) {
+  return pako.deflate(uncompressed);
+}
+
+function processDataArray(
+  dataArray,
+  format,
+  blockSize,
+  compressor = 'vtkZLibDataCompressor'
+) {
+  if (format === FormatTypes.ASCII) {
+    return dataArray.getData().join(' ');
+  }
+  if (format === FormatTypes.BINARY) {
+    if (compressor === 'vtkZLibDataCompressor') {
+      // ----------------------------------------------------------------------
+      // Layout of the data
+      // header[N, s1, s1, blockSize1, ..., blockSizeN], [padding???], block[compressedData], ..., block[compressedData]
+      // [header] N, s1 and s2 are uint 32 or 64 (defined by header_type="UInt64" attribute on the root node)
+      // [header] s1: uncompress size of each block except the last one
+      // [header] s2: uncompress size of the last blocks
+      // [header] blockSize: size of the block in compressed space that represent to bloc to inflate in zlib. (This also give the offset to the next block)
+      // ----------------------------------------------------------------------
+
+      const componentUint8Size = dataArray.getElementComponentSize();
+      const uncompressedUint8Size =
+        dataArray.getNumberOfValues() * componentUint8Size;
+      const blockUint8Size = Math.min(blockSize, uncompressedUint8Size);
+      const nbBlocks = Math.ceil(uncompressedUint8Size / blockUint8Size);
+      const headerUint8Size = 4 * (3 + nbBlocks);
+      let totalUint8Length = headerUint8Size;
+      const header = new Uint32Array(headerUint8Size / 4);
+      header[0] = nbBlocks; // N
+      header[1] = blockUint8Size; // s1
+      header[2] = uncompressedUint8Size - nbBlocks * blockUint8Size; // s2
+      if (header[2] === 0) {
+        header[2] = blockUint8Size;
+      }
+      const blocks = [];
+      let dataOffset = 0;
+      for (let blockId = 0; blockId < nbBlocks; ++blockId) {
+        const uncompressedBlock = new Uint8Array(
+          dataArray.getData().buffer,
+          dataOffset,
+          blockUint8Size
+        );
+        dataOffset += blockUint8Size;
+        const compressedUint8Block = compressBlock(uncompressedBlock);
+        blocks.push(compressedUint8Block);
+        header[3 + blockId] = compressedUint8Block.length;
+        totalUint8Length += compressedUint8Block.length;
+      }
+      const uint8 = new Uint8Array(totalUint8Length);
+      let uint8Offset = 0;
+      const headerUint8 = new Uint8Array(header.buffer);
+      uint8.set(headerUint8, uint8Offset);
+      uint8Offset += headerUint8Size;
+      for (let blockId = 0; blockId < nbBlocks; ++blockId) {
+        uint8.set(blocks[blockId], uint8Offset);
+        uint8Offset += header[3 + blockId];
+      }
+      return fromByteArray(uint8);
+    }
+    return 'Only vtkZLibDataCompressor is supported';
```

Comment:
Better to `throw new Error('...message...')` than return your error message.

Same with line 78 and 80.